### PR TITLE
feature(build): Integrate `garden-repo-manager`

### LINKED ISSUE #1433: refactor `garden-debian-sources-list` script
**What would you like to be added**:
While integrating a new native Debian mirror support for Garden Linux, we encountered several things that should be adjusted within the `garden-debian-sources-list` script. However, we probably should completely refactor this one after #1426 got merged.

There shouldn't be any `--gardenlinux-only` param anymore since this is our default. Next, we probably can remove older references of `squeeze` etc. With #1426 the orchestrating scripts are already adjusted and the handling of a non existent `--gardenlinux-only` param integrated. Therefore, we only may call it with an additional `--debian-mirror` param which adds the native Debian repository.

### LINKED ISSUE #1470: `Build Pipeline`: crossbuilding may not work any more
**What happened:**
Currently, crossbuilding may not work anymore due to some issues within the build pipeline.

Reminder for me; self assigned.

### LINKED ISSUE #1473: `Build Pipeline`: local package support is broken
**What happened:**
Currently, the local package support for installing .deb files from a `packages` directory is broken. While #1472 introduces a new handling of handling repositories (including local package support) the Makefile and pipeline must be adjusted.

### PR DESCRIPTION
**Pull request by gyptazy**
Integrate `garden-repo-manager` for repository management

**How to categorize this PR?**

/kind cleanup
/area os
/os garden-linux

**What this PR does / why we need it**:
 * Integrate `garden-repo-manager` for repository management
 * Fix crosscompile in `arm64` (to create `amd64`)
 * Fix `local_packages` support

**Which issue(s) this PR fixes**:
Fixes #1433
Fixes #1470 
Fixes #1473

**Special notes for your reviewer**:
This is based on the new `garden-repo-manager` package for Garden Linux.
See also: https://gitlab.com/gardenlinux/gardenlinux-package-gardenlinux-repo-manager

**Release note**:

```feature user
- category: feature
- target_group: user, operator, developer
```

**WIP**

### Default
Within the default case, only the Garden Linux repository will be added:
```
# cat /etc/apt/sources.list.d/GRM_gardenlinux_today.sources 
X-Repolib-Name: Managed_gardenlinux_today
Enabled: yes
URIs: https://repo.gardenlinux.io/gardenlinux
Types: deb
Architectures: amd64
Suites: today
Components: main
Signed-By: /etc/apt/trusted.gpg.d/gardenlinux.asc
```

### Debian Mirror
Next to the default, Garden Linux provides the way to add Debian repositories at build time by defining `
--debian-mirror`. This adds next to the Garden Linux repository the Debian ones:
```
# ls /etc/apt/sources.list.d/GRM_*
/etc/apt/sources.list.d/GRM_debian_testing.sources
/etc/apt/sources.list.d/GRM_gardenlinux_today.sources
```

Configuration of repo metadata:
```
# cat /etc/apt/sources.list.d/GRM_*
X-Repolib-Name: Managed_debian_testing
Enabled: yes
URIs: https://deb.debian.org/debian
Types: deb
Architectures: amd64
Suites: testing
Components: main contrib

X-Repolib-Name: Managed_gardenlinux_today
Enabled: yes
URIs: https://repo.gardenlinux.io/gardenlinux
Types: deb
Architectures: amd64
Suites: today
Components: main
Signed-By: /etc/apt/trusted.gpg.d/gardenlinux.asc
```

### Local packages as Mirror
Additionally, Garden Linux provides possibilities to inject local packages in `.deb` fomat to its artifacts by creating a `packages` folder and placing `.deb` files in it. However, this was tested during the creation of `garden-repo-manager` itself but is currently broken within the Garden Linux build pipeline. As a result, we should currently ignore this case which'll be fixed in https://github.com/gardenlinux/gardenlinux/issues/1473 and is unrelated to the introduction of the `garden-repo-manager` itself.

### CODE REVIEW

**Comment by gyptazy** [comment]
Ready for review. Checks passed.

File: bin/garden-build
Comment (nanory) [inline bin/garden-build:184]: This will not work due to the single quotes

**Review by nanory — CHANGES_REQUESTED** [review]

File: bin/garden-build
Comment (gyptazy) [reply bin/garden-build]: oops, copy paste at its best. thanks... :)

**Review by nanory — APPROVED** [review]

**Comment by gyptazy** [comment]
Before merging, someone of the Garden Linux core team should review this change. However, the dedicated test cases can be found within the first post.

cc: @gehoern @Vincinator

**Review by gehoern — APPROVED** [review]

**Pull request merged**
